python/dgl/distributed/graph_partition_book.py

```diff
@@ -0,0 +1,250 @@
+"""Define graph partition book."""
+
+import json
+import numpy as np
+
+from .. import backend as F
+from ..base import NID, EID
+from ..data.utils import load_graphs
+
+class GraphPartitionBook:
+    """Partition information.
+
+    Note that, we assume that all partitions exists in local machines for now.
+    Once we have the reshuffle() api we can change that.
+    """
+    def __init__(self, partition_config_file, ip_config_file):
+        """Initialization
+
+        Parameters
+        ----------
+        partition_config_file : str
+            path of graph partition file.
+        ip_config_file : str
+            path of IP configuration file.
+        """
+        with open(partition_config_file) as conf_f:
+            self._part_meta = json.load(conf_f)
+        # Read ip list from ip_config_file
+        self._ip_list = []
+        lines = [line.rstrip('\n') for line in open(ip_config_file)]
+        # we assume that ip is sorted by machine ID in ip_config_file
+        for line in lines:
+            ip_addr, _, _ = line.split(' ')
+            self._ip_list.append(ip_addr)
+        # Get number of partitions
+        assert 'num_parts' in self._part_meta, "cannot get the number of partitions."
+        self._num_partitions = self._part_meta['num_parts']
+        assert self._num_partitions > 0, 'num_partitions cannot be a negative number.'
```

Comment:
There are many code duplicates from here: https://github.com/dmlc/dgl/blob/master/python/dgl/distributed/partition.py#L123 . I wonder can we:
(1) have this `GraphPartitionBook` support save/load to/from JSON?
(2) reuse it in `dgl.distributed.partition.load_partition` and `dgl.distributed.partition.partition_graph`?